### DISCUSSION TITLE
Set a default background color

### DISCUSSION THREAD

**Discussion by KiruPoruno** [community/community#142742] [npm]
Currently there's no default background color set on various elements, notably the search results, however the text color is set, meaning if you've a browser or extension that sets this for whatever reason, it may cause it to be unreadable.

As an example, if the browser or an extension sets the default background color to `black` you get unreadable search results:

![npm](https://github.com/user-attachments/assets/7f9bb011-8e8f-4d27-87e5-a66e63e664e2)
![test - npm](https://github.com/user-attachments/assets/5df38634-2912-4564-acec-e1575ed6ee58)

Notably however, this only applies until you start scrolling:

![test - npm search](https://github.com/user-attachments/assets/5d4decc2-b53c-495f-9bae-75b233e41da3)

This is a very easy fix (`html, body {background: white}`), and it's been bugging me for years.

**Comment by leobalter** [comment]
would you mind sharing what browser are you using? We do have a root background set:

```css
:root {
    background-color: #fff;
    background-color: var(--bg);
}
```

I'm updating it to use the most current variable but I believe your browser must support `:root` regardless.

**Reply by KiruPoruno** [reply]
Happens in any browser really, the issue actually stems from only applying it to `:root`/`html` and not also `body`, as the color is then below `body`, so if `body` then gets a color, it appears above it.

Edit: To be clear, this is also why the color fixes itself when scrolling down, as the body is only `100vh` tall, and so the `:root`/`html` color becomes visible.

**Reply by leobalter** [reply]
I'd appreciate it if you can help me with steps to reproduce this issue.

**Reply by KiruPoruno** [reply]
Apply a color to the `body` in some form, and you'll see the above, i.e `body {background: black}`, for reference, this isn't an issue for most people, but some dark mode extensions do this, and a couple browsers also do this directly, which is where the problem occurs.

**Reply by leobalter** [reply]
> Apply a color to the body in some form

Ok, I need something more general here. Is this a configuration coming from the browser? Is this an extension applying custom colors to a page?

Giving you an example: I tried going to Firefox `about:config` and changing the `browser.display.background_color` to a custom color (let's say `#000`). That was not enough to reproduce the issue as the background remained consistent.

I also tried going to Chrome DevTools and turning on the "Automatic Dark Mode" option. Everything worked just fine without mixing up.

If can only provide me an example were you directly - and intentionally - change the body color, I'm not able to support this as not a bug.

**Reply by KiruPoruno** [reply]
I recall qutebrowser's `colors.webpage.bg` option also causing this at one point, but it seems to no longer be an issue in its regard, in my own case, it was a custom stylesheet which has an issue with this (this has since been manually fixed a while ago, however every so often I discover it again on some device or browser), I also recall having this problem at one point on mobile (likely a Chromium fork, some months ago...) with a similar issue (can't reproduce anymore).

This is a very minor issue, and affects very few people, but figured it might've been worth reporting anyway... :p
